File: src/main/java/de/pfleghar/shoppinglist/backend/service/ShoppingService.java
package de.pfleghar.shoppinglist.backend.service;

import de.pfleghar.shoppinglist.backend.domain.ShoppingItem;

import java.util.List;

public interface ShoppingService {

    List<ShoppingItem> getItems();

    void addItem(ShoppingItem item);

    void updateItem(ShoppingItem item);

    void deleteItem(long id);

    void clearShoppingList();
}

File: client/src/app/shared/shopping/shopping.service.ts
import { Injectable } from '@angular/core';
import { HttpClient} from '@angular/common/http';
import { Observable} from 'rxjs/index';

@Injectable({
  providedIn: 'root'
})
export class ShoppingService {

  constructor(private http: HttpClient) { }

  getItems(): Observable<any> {
    return this.http.get('//localhost:8081/getItems');
  };

  addItem(shoppingItem: any){
    this.http.post('//localhost:8081/addItem', shoppingItem);
  }

  updateItem(shoppingItem: any) {
    this.http.post('//localhost:8081/updateItem', shoppingItem);
  }

  removeItem(itemId: number): Observable<any> {
    return this.http.delete('//localhost:8081/removeItem?id=' + itemId);
  }

  removeAll(): Observable<any> {
    return this.http.delete('//localhost:8081/clearShoppingList');
  }

}
